File: CodingSamples/Java/Language/ClassTypeTest/Investment.java
class Investment {

	//instance fields - values will be separately stored in each instance
	private double installment;
	private int years;
	private boolean risk;

	//constructor - initializes new instance, if no constructor is explictly
	//defined in a class a parameterless constructor is implicitly define
	public Investment(double amount, int period) {
		installment = amount;
		years = period;
		risk = false;
	}

	//instance method - can only be called on object using . operator,
	//receives a hidden 'this' argument which refers to the object
	//on (left side of . operator) which this method is called 
	public void allowRisk(boolean yes) {
		risk = yes; //this.risk = yes
	}

	public double totalAmount() {
		return installment * years;
	}

	public double futureValue() {
		float i = risk ? 0.11f : 0.06f;
		return (installment / i) * (Math.pow(1 + i, years) - 1);
	}

}

File: CodingSamples/DotNet/Language/StructTypeTest/DemoApp/Investment.cs
//user-defined value type
struct Investment
{
    //instance fields
    private double installment;
    private int period;
    private bool risk;

    //constructor - method for initializing a new instance
    public Investment(double amount, int years)
    {
        installment = amount;
        period = years;
        risk = false;
    }

    //instance (non-static) method must be called
    //on an object of its type (using . operator) and
    //it can refer to other instance members through
    //the hidden 'this' argument which refers to the 
    //object on which it is called 
    public void AllowRisk(bool yes)
    {
        risk = yes; //this.risk = yes
    }

    public double TotalAmount()
    {
        return installment * period;
    }

    public double FutureValue()
    {
        float i = risk ? 0.11f : 0.06f;
        return (installment / i) * (Math.Pow(1 + i, period) - 1);
    }
}
